### ISSUE TITLE
Case unsensitive unique constraint

### ISSUE THREAD

**Issue by Mohammed-Farid** [prisma/prisma#3851]
## Problem
It's really important to keep the casing in some of our columns which are unique ( cannot be duplicated ) which constraint can be overridden so easily by just change the casing

## Suggested solutions
I've found some articles and the most matching one is [this](http://shuber.io/case-insensitive-unique-constraints-in-postgres/)
Within schema it'd be great having 
```prisma
model Station {
	...
	uniqueField String @unique(mode:"insensitive")
}
```

## Alternatives
The second solution which is the only available at the moment is to create another column with lowercased letters and it'll be the unique one

**Comment by pantharshit00** [comment]
This is more likely to be solved by `citext` support in schema. That will come eventually via https://github.com/prisma/prisma/issues/3447 although since citext is an extension instead of built in postgres type, it can take more time.


After that you could possibiliy do:
```prisma
model Blog {
  ...
  name String @pg.Citext @unique
}
```

**Comment by probablycorey** [comment]
It doesn't look like #3447 supports a citext db type for postgres. Are there any plans to support the functionality that @Mohammed-Farid described?

**Comment by albertoperdomo** [comment]
@probablycorey Yes, we are adding support Citext as a type. This is scheduled for 2.16.0. I will also update #3447.

**Comment by albertoperdomo** [comment]
@Mohammed-Farid Does using Citext work for you as a solution?

**Comment by Mohammed-Farid** [comment]
@albertoperdomo Yes it should fix it as @pantharshit00 has shown previously
